---
title: "Slueth"
author: "Lori K. Bogren"
date: "1/15/2016"
output: html_document
---
Slueth:
http://pachterlab.github.io/sleuth/

A sleuth for RNA-Seq begins with the quantification of samples with kallisto, and together a sleuth of kallistos can be used to analyze RNA-Seq data rigorously and rapidly.

https://liorpachter.wordpress.com/2015/08/17/a-sleuth-for-rna-seq/

```{r}
#To install sleuth start R and first install rhdf5 by typing:

source("http://bioconductor.org/biocLite.R")
biocLite("rhdf5")

#Then install devtools by typing

install.packages("devtools")

#and install sleuth by typing

devtools::install_github("pachterlab/sleuth")

#Next load sleuth with

library("sleuth")
```


The first step in a sleuth analysis is to specify where the kallisto results are stored. 

Begin by storing the base directory of the results in a variable (base_dir),

```{r}
base_dir <- "/vol3/home/bogrenl/projects/RNASeq/kallisto-output-bootstrap-formatted"

```

Next get the list of sample IDs with

```{r}

sample_id <- dir(file.path(base_dir,"results"))

```

The result can be displayed by typing

```{r}

sample_id

```

A list of paths to the kallisto results indexed by the sample IDs is collated with

```{r}

kal_dirs <- sapply(sample_id, function(id) file.path(base_dir, "results", id, "kallisto"))
kal_dirs

```

The next step is to load an auxillary table that describes the experimental design and the relationship between the kallisto directories and the samples:

```{r}

s2c <- read.table(file.path(base_dir, "BrainRNAseqMetadata-1.txt"), header = TRUE, stringsAsFactors=FALSE)
s2c <- dplyr::select(s2c, sample = run_accession, condition)
s2c

```

Now, we must enter the directories into a column in the table describing the experiment. This column must be labeled path, otherwise sleuth will throw an error. This is to ensure that the user can check which samples correspond to which kallisto runs.

```{r}

s2c <- dplyr::mutate(s2c, path = kal_dirs)

```

The user should check whether or not the order is correct. In this case, the kallisto output is correctly matched with the sample identifiers.

Make sure to do this --> they are not automatically lined up.  You may need to edit the metadata sample order to match the filepath sample order.

```{r}

print(s2c)

```

Now the “sleuth object” can be constructed. This requires three commands that (1) load the kallisto processed data into the object (2) estimate parameters for the sleuth response error measurement model and (3) perform differential analysis (testing). On a laptop the three steps should take about 2 minutes altogether.

First type

```{r}

so <- sleuth_prep(s2c, ~ condition)

```

then

```{r}

so <- sleuth_fit(so)

```

and finally

```{r}


so <- sleuth_wt(so, 'conditionLT')
#so <- sleuth_wt(so, 'conditionscramble')


#have to specify which condition --> looks like you can only compare two contisions at once.  The intercept is chosen based on alphbetical order or the condition column ...in our data set it would be Ar.

```

In general, one can see the possible tests that could be performed using the which_beta parameter in sleuth_wt and examining the coefficients:

```{r}

models(so)

```
output:
> models(so)
[  full  ]
formula:  ~condition 
coefficients:
  (Intercept)
 	conditionEnt
 	conditionIBA
 	conditionLT
 	conditionSA
 	conditionSpD


At this point the sleuth object constructed from the kallisto runs has information about the data, the experimental design, the kallisto estimates, the model fit, and the testing. In other words it contains the entire analysis of the data. There is, however, one piece of information that can be useful to add in, but that is optional. In reading the kallisto output sleuth has no information about genes, but this can be added allowing for searching and analysis by gene instead of transcript.

Since the example was constructed with the ENSEMBL human transcriptome, we will add gene names from ENSEMBL using biomaRt (there are other ways to do this as well):

First, install biomaRt with

```{r}

source("http://bioconductor.org/biocLite.R")
biocLite("biomaRt")


```

Then collect gene names with

```{r}

mart <- biomaRt::useMart(biomart = "ensembl", dataset = "itridecemlineatus_gene_ensembl")

```

and add them into the sleuth table with

```{r} 

#This is not working.  The target_id do not match the ensemble transcript IDs....

t2g <- biomaRt::getBM(attributes = c("ensembl_transcript_id", "ensembl_gene_id", "external_gene_name"), mart = mart)
t2g <- dplyr::rename(t2g, target_id = ensembl_transcript_id,ens_gene = ensembl_gene_id, ext_gene = external_gene_name)
so <- sleuth_prep(s2c, ~ condition, target_mapping = t2g)

so <- sleuth_fit(so)


so <- sleuth_wt(so, which_beta = 'conditionLT')

```

This addition of metadata to transcript IDs is very general, and can be used to add in other information.

The best way to view the results is to generate the Shiny webpage that allows for exploratory data analysis:

```{r}

sleuth_live(so) #This doesn't work on Tesla to pull up a browser window.  Have to run on laptop
```

To generate a table of results for analysis within R type

```{r}

results_table2 <- sleuth_results(so, 'conditionLT')

write.table(results_table, file="Sleuth_test_results_table-012516.txt")



plot_transcript_heatmap(so, transcripts, units = "tpm", trans = "log",offset = 1)

```

